lib/core/uploaderError.js

```diff
@@ -17,29 +17,43 @@
 
 var sundial = require('sundial');
 
-// Create a new object, that prototypally inherits from the Error constructor.
-function UploaderError(message, stage, error) {
-
-  this.name = this.constructor.name;
-  this.message = message;
+// if another error is passed in then attach it
+function attachOriginal(otherError){
+  if(otherError){
+    return { 'originalError' : otherError };
+  }
+  return;
+}
 
+// creates our debug info from the details passed
+function buildDebug(message, stage, otherError){
   var theStep = stage.Step || '';
-  if(error && error.step){
-    theStep = error.step;
+  if(otherError && otherError.step){
+    theStep = UploaderError.STEP +' '+otherError.step;
+  } else if (theStep) {
+    theStep = UploaderError.STEP +' '+theStep;
   }
 
-  this.debug = [
-    message || stage.Message,'|',
+  return [
+    message || otherError.message ,'|',
     UploaderError.UTC_TIME, sundial.utcDateString(),'|',
     UploaderError.CODE, stage.Code,'|',
-    UploaderError.STEP, theStep
+    UploaderError.CODE_MSG, stage.Message,'|',
+    theStep
```

Comment:
So you will see it used in the tests but it might be a redundant feature? The intention was if you wanted to attach step information from a driver where you either catch or throw an error that would be meaningful in diagnosing what went wrong then this is a way to do that.